### PR TITLE
Disable next button for sessions when unanswered.

### PR DESCRIPTION
**Pull request by eschlange**
[#101203814]

### CODE REVIEW

File: src/js/controllers/SessionsController.js [hunk 1]
```diff
@@ -20,6 +20,7 @@
         question: this.currentQuestion,
         answer: this.currentAnswer
       });
+      this.currentAnswer = undefined;
```
Comment (ericcf) [inline src/js/controllers/SessionsController.js]: Why did you decide to initialize to `undefined` instead of `null`?

Comment (michaelwehrley) [inline src/js/controllers/SessionsController.js]: How come you need to specify that it is `undefined`?

File: test/unit/controllers/SessionsControllerSpec.js [hunk 2]
```diff
@@ -114,4 +114,22 @@ describe('SessionsController', function () {
       expect(controller.getSessionContent()).to.equal('some_content');
     });
   });
+
+  describe('#disableNext', function() {
```
Comment (ericcf) [inline test/unit/controllers/SessionsControllerSpec.js]: This spec looks good, but there are no specs for `#saveSessionAnswer` or `#next` on this controller.

**Comment by eschlange** [comment]
@ericcf @michaelwehrley Updated the controller to check for nulls, didn't know that it was bad practice to use undefined until now. 

Also added specs for some of the older methods.

**Comment by ericcf** [comment]
One argument against initializing a property to `undefined` is that it becomes ambiguous as to whether the property is uninitialized or initialized to `undefined`:

```javascript
a = {};
a.foo = undefined;
a.foo === undefined; // true
b = {};
b.foo === undefined; // true
a.foo === b.foo; // true
```

Another thing to note:

```javascript
a = {};
a.foo === undefined; // true
a.foo == null; // true
a.foo === null; // false !
```

**Comment by michaelwehrley** [comment]
I was wondering if that was part of the argument.  Thanks for the example and clarification.

**Pull request merged**